src/test/java/pl/techlab24/OSKManager/model/validation/DriverCandidateProfileValidatorTest.java

```diff
@@ -0,0 +1,132 @@
+    }
+
+    @Test
+    void shouldValidateNullDriverCandidateProfile() {
+        List<String> resultOfValidation = DriverCandidateProfileValidator.validate(null);
+        assertEquals(Collections.singletonList("Driver candidate profile cannot be null."), resultOfValidation);
+    }
+
+    @ParameterizedTest
+    @MethodSource("setOfNumbersAndValidationResults")
+    void shouldValidateNumber(String number, List<String> expected) {
+        DriverCandidateProfile driverCandidateProfileWithVariableNumber =
+            correctDriverCandidateProfile.toBuilder().number(number).build();
+
+        List<String> resultOfValidation = DriverCandidateProfileValidator.validate(driverCandidateProfileWithVariableNumber);
+
+        assertEquals(expected, resultOfValidation);
+    }
+
+    private static Stream<Arguments> setOfNumbersAndValidationResults() {
+        return Stream.of(
+            Arguments.of(null, Collections.singletonList("Driver candidate profile number cannot be null.")),
+            Arguments.of("", Collections.singletonList("Driver candidate profile number must contain at least 1 character.")),
+            Arguments.of("   ", Collections.singletonList("Driver candidate profile number must contain at least 1 character.")),
+            Arguments.of("Test number", Collections.emptyList()),
+            Arguments.of("AAA-1", Collections.emptyList()),
+            Arguments.of("* test 0123", Collections.emptyList())
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("setOfCategoriesAndValidationResults")
+    void shouldValidateCategory(Category category, List<String> expected) {
+        DriverCandidateProfile driverCandidateProfileWithVariableCategory =
+            correctDriverCandidateProfile.toBuilder().category(category).build();
+
+        List<String> resultOfValidation = DriverCandidateProfileValidator.validate(driverCandidateProfileWithVariableCategory);
+
+        assertEquals(expected, resultOfValidation);
+    }
+
+    private static Stream<Arguments> setOfCategoriesAndValidationResults() {
+        Category correctCategory = Category.builder()
+            .id(1L)
+            .categoryName("Test category")
+            .build();
+
+        Category incorrectCategoryWithNullCategoryName = Category.builder()
+            .id(1L)
+            .categoryName(null)
+            .build();
+
+        Category incorrectCategoryWithEmptyCategoryName = Category.builder()
+            .id(1L)
+            .categoryName("")
+            .build();
+
+        return Stream.of(
+            Arguments.of(null, Collections.singletonList("Category cannot be null.")),
+            Arguments.of(correctCategory, Collections.emptyList()),
+            Arguments.of(incorrectCategoryWithNullCategoryName, Collections.singletonList("Category name cannot be null.")),
+            Arguments.of(incorrectCategoryWithEmptyCategoryName, Collections.singletonList("Category name must contain at least 1 character."))
+        );
+    }
+
```

Comment:
move this method's body to private static class SetsOfVariablesAndValidationResults, because the same method we can find in CourseValidatorTest class

Reply:
Moved to  final class SetsOfVariablesAndValidationResults.